apps/studio/components/interfaces/HomeNew/GettingStarted/useGettingStartedProgress.ts

```diff
@@ -0,0 +1,147 @@
+import { PROJECT_STATUS } from 'lib/constants'
+
+type GettingStartedStatuses = {
+  hasTables: boolean
+  hasCliSetup: boolean
+  hasSampleData: boolean
+  hasRlsPolicies: boolean
+  hasConfiguredAuth: boolean
+  hasAppConnected: boolean
+  hasFirstUser: boolean
+  hasStorageObjects: boolean
+  hasEdgeFunctions: boolean
+  hasReports: boolean
+  hasGitHubConnection: boolean
+}
+
+export const useGettingStartedProgress = (): GettingStartedStatuses => {
+  const { data: project } = useSelectedProjectQuery()
+  const { data: organization } = useSelectedOrganizationQuery()
+
+  const projectRef = project?.ref
+  const connectionString = project?.connectionString
+  const isProjectActive = project?.status === PROJECT_STATUS.ACTIVE_HEALTHY
+
+  const { data: tablesData } = useTablesQuery(
+    { projectRef, connectionString, schema: 'public' },
+    { enabled: !!projectRef && isProjectActive }
+  )
+
+  const { data: storageTablesData } = useTablesQuery(
+    { projectRef, connectionString, schema: 'storage' },
+    { enabled: !!projectRef && isProjectActive }
+  )
+
+  const { data: policiesData } = useDatabasePoliciesQuery(
+    { projectRef, connectionString, schema: 'public' },
+    { enabled: !!projectRef && !!connectionString && isProjectActive }
+  )
+
+  const { data: authConfig } = useAuthConfigQuery(
+    { projectRef },
+    { enabled: !!projectRef && isProjectActive }
+  )
+
+  const { data: usersCountData } = useUsersCountQuery(
+    { projectRef, connectionString },
+    { enabled: !!projectRef && !!connectionString && isProjectActive }
+  )
+
+  const { data: edgeFunctionsData } = useEdgeFunctionsQuery(
+    { projectRef },
+    { enabled: !!projectRef && isProjectActive }
+  )
+
+  const { data: reportsData } = useContentInfiniteQuery(
+    { projectRef, type: 'report', limit: 1 },
+    { enabled: !!projectRef && isProjectActive }
+  )
+
+  const { data: migrationsData } = useMigrationsQuery(
+    { projectRef, connectionString },
+    { enabled: !!projectRef && !!connectionString && isProjectActive }
+  )
+
+  const { data: usageStatsData } = useProjectLogStatsQuery(
+    { projectRef, interval: '1day' },
+    { enabled: !!projectRef && isProjectActive }
+  )
+
+  const { data: githubConnections } = useGitHubConnectionsQuery(
+    { organizationId: organization?.id },
+    { enabled: !!projectRef && !!organization?.id }
+  )
+
+  const statuses = useMemo<GettingStartedStatuses>(() => {
+    const hasTables = (tablesData?.length ?? 0) > 0
+    const hasCliSetup = (migrationsData?.length ?? 0) > 0
+    const hasSampleData = (tablesData ?? []).some(
+      (table) => Number(table?.live_rows_estimate ?? 0) > 0
+    )
+    const hasRlsPolicies = (policiesData?.length ?? 0) > 0
+    const allowSignupsEnabled = authConfig ? !authConfig.DISABLE_SIGNUP : false
+    const emailProviderEnabled = !!authConfig?.EXTERNAL_EMAIL_ENABLED
+    const hasConfiguredAuth = allowSignupsEnabled && emailProviderEnabled
```

Comment:
This check will return true for all new projects btw - if it might be an issue

<img width="604" height="538" alt="image" src="https://github.com/user-attachments/assets/3b1e2935-1090-4352-9aad-9a099096b2da" />